package/MDAnalysis/topology/MOL2Parser.py

```diff
@@ -150,8 +152,28 @@ def parse(self, **kwargs):
         charges = []
 
         for a in atom_lines:
-            aid, name, x, y, z, atom_type, resid, resname, charge = a.split()[:9]
-
+            columns = a.split()
+            if len(columns) >= 9:
+                columns = columns[:9]
+                aid, name, x, y, z, atom_type, resid, resname, charge = columns
+            else:
+                opt_values = ['1', 'UNK', '0.0']
+                opt_fields = ['subst_id', 'subst_name', 'charge']
```

Comment:
I think optional values should not result in attributes, but should be discarded completely if not available. With this solution, you will end up with a `charges` attribute with all `0.0`, for example.